src/ui/DynamicFacet/DynamicFacetValues/DynamicFacetValue.ts

```diff
@@ -1,18 +1,24 @@
 import * as Globalize from 'globalize';
-import { DynamicFacetValueRenderer } from './DynamicFacetValueRenderer';
-import { DynamicFacet } from '../DynamicFacet';
 import { FacetValueState } from '../../../rest/Facet/FacetValueState';
 import { IAnalyticsDynamicFacetMeta, analyticsActionCauseList } from '../../Analytics/AnalyticsActionListMeta';
 import { l } from '../../../strings/Strings';
 import { IRangeValue, RangeType } from '../../../rest/RangeValue';
 import { FacetType } from '../../../rest/Facet/FacetRequest';
+import { IDynamicFacet } from '../DynamicFacetInterface';
+import { IFacetResponseValue } from '../../../rest/Facet/FacetResponse';
+
+export interface ValueCreator {
+  createFromResponse(facetValue: IFacetResponseValue, index: number): DynamicFacetValue;
+  createFromValue(value: string): DynamicFacetValue;
+  createFromRange?(range: IRangeValue, index: number): DynamicFacetValue;
```

Comment:
Hmm, why is the `createFromRange` method optional?

Reply:
Because a Specific facet wont create from a range

Reply:
Okay, in that case we should have a separate interface that extends `IValueCreator` and adds the method.

Reply:
Exactly what Sami said. A good interface is a concise as it needs to, and more complex interface can be constructed by composing multiple together.

Reply:
Not sure it's going to work that simply here (referring to the DynamicFacetValues file) but I'll try something